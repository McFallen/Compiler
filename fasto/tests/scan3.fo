fun bool and(bool a, bool b) = a && b
fun bool or(bool a, bool b) = a || b

fun bool writeBool(bool n) = write(n)

fun [bool] main() =
  let a = {true, true, true, true} in
  let b = scan(or, true, a) in
  map(writeBool, b)