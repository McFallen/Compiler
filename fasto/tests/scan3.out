False False False True True